### ISSUE TITLE
Maths.ceilN limitation?

### ISSUE THREAD

**Issue by scottsue** [OpenHFT/Chronicle-Core#80]
Hi,

I'm just testing our the Maths function to perform various levels of rounding and to see what the limitations are (if any) as I want to avoid the usage of BigDecimals.

I've come across a use case where I have a large number of decimals (64.09159469999999).  When running this number through the Maths.ceil to 7 digits, I get returned the same number

Running the quick test below 
```
public class TestCeiling {
	public static void main(String[] args) {
		for (int i = 0; i < 10; i++) {
			double roundedDouble = Maths.floorN(64.09159469999999, i);
			System.out.println("To precision " + i +" - roundedDouble = " + roundedDouble);
		}
	}
}
```
I get the following output
```
To precision 0 - roundedDouble = 64.0
To precision 1 - roundedDouble = 64.0
To precision 2 - roundedDouble = 64.09
To precision 3 - roundedDouble = 64.091
To precision 4 - roundedDouble = 64.0915
To precision 5 - roundedDouble = 64.09159
To precision 6 - roundedDouble = 64.091594
To precision 7 - roundedDouble = 64.09159469999999
To precision 8 - roundedDouble = 64.09159469999999
To precision 9 - roundedDouble = 64.09159469999999
```

Looking at the underlying function, I suspect this is due to the '+8' on the number of digits that I want to round to to get the factor.

Interestingly enough, when I run the 'false' portion of the calculation 
```
Math.floor(Math.round(d * factor) / 1e8)
```
I get the expected result that I'm looking for.  Is this something expected as a limitation of the library considering we're dealing with doubles here?

Thanks in advance!

**Comment by scottsue** [comment]
Hi sorry to resurrect this.  However, is this something simple that I'm missing or is this a genuine limitation?

**Comment by scottsue** [comment]
Hi is there any movement on this or thoughts?

**Comment by peter-lawrey** [comment]
Note: These methods assume there could be some rounding error. The rounding error is now assumed to be 1 ulp.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.20.40](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.20.40), [BOM-2.20.60](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.60)

**Comment by scottsue** [comment]
Great thanks.  Look forward to testing this out

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.20.102](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.20.102), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
